codegen/value.go

```diff
@@ -138,8 +139,9 @@ func (v *zeroValue) Kind() parser.Kind {
 
 func (v *zeroValue) Filesystem() (Filesystem, error) {
 	return Filesystem{
-		State: llb.Scratch(),
-		Image: &solver.ImageSpec{},
+		State:    llb.Scratch(),
+		Image:    &solver.ImageSpec{},
+		Platform: specs.Platform{OS: runtime.GOOS, Architecture: runtime.GOARCH},
```

Comment:
Not sure if `ZeroValue` is used for anything in practice, but if it is, I'm wondering if this may cause some problems down the line. We may be trying to build an image for a different platform than the host.

It looks like `scratch` doesn't involve this code path, so it's probably not an issue... Makes sense to try it and see.

Reply:
Yeah, I don't think the ZeroValue with FileSystem is practically used.  Also in order for this Platform to be used, it would have to be the only `fs` in context, which is also probably not practically ever used.  I could leave this as an empty `specs.Platform{}` if you think that makes more sense, perhaps it would be better to have an empty Platform spec than the default one in the case we accidentally use this `Platform` ...

Reply:
I have updated to leave this Platform empty, I dont see how it could be used, probably better to get an error in the future if we actually try to this empty Platform in the future.

Reply:
LGTM